### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by danorris709**

### CODE REVIEW

File: GameArena.java [hunk 1]
```diff
@@ -282,22 +282,13 @@ public void paint (Graphics gr)
 	// 
 	private Color getColourFromString(String col)
 	{
-		Color c = colours.get(col.toUpperCase());
-
-		if (c == null && col.startsWith("#"))
-		{
+		return colours.computeIfAbsent(col, ___ -> {
```
Comment (finneyj) [inline GameArena.java]: I think you might have erased a case sensitivity check here.

File: GameArena.java [hunk 2]
```diff
@@ -282,22 +282,13 @@ public void paint (Graphics gr)
...
 
-			c = new Color(r,g,b);
-			colours.put(col.toUpperCase(), c);
-		}
-
-		if (c == null)
-			c = Color.WHITE;
-
-		return c;
+			return new Color(r, g, b);
```
Comment (finneyj) [inline GameArena.java:306]: this also doesn't repopulate the map with the new colour does it?

Comment (danorris709) [reply GameArena.java:306]: computeIfAbsent fills the map with what's returned from the lambda if it's not already there (and subsequently returns it)

File: GameArena.java [hunk 3]
```diff
@@ -466,8 +466,19 @@ public void removeText(Text t)
...
+		this.pause(20);
+	}
+
+	/**
+	 * 
+	 * Pause the {@link Thread} for {@param millis} milli-seconds
+	 * 
+	 * @param millis the number of milli-seconds the {@link Thread} is paused for
+	 */
+	public void pause(int millis) {
```
Comment (finneyj) [inline GameArena.java]: would be better as a accessor/mutator pair to request and obtain a frame rate?

File: GameArena.java [hunk 4]
```diff
@@ -115,7 +119,7 @@ private void init(int width, int height, boolean createWindow)
 		renderingHints.put(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_LCD_HRGB);
 
 		try {
-			Class rh = Class.forName("java.awt.RenderingHints");
+			Class<?> rh = Class.forName("java.awt.RenderingHints");
```
Comment (finneyj) [inline GameArena.java:138]: Which JDK did this raise a warning on?

File: GameArena.java (hunk 3 above)
Comment (danorris709) [reply GameArena.java]: Not quite sure what you mean by this?

File: GameArena.java (hunk 4 above)
Comment (danorris709) [reply GameArena.java:138]: I believe the VS Code extension runs Java 8. However, I'm not certain about that

File: GameArena.java (hunk 3 above)
Comment (danorris709) [reply GameArena.java]: Oh okay, I understand what you mean now. Yeah that sounds interesting I can sort that out

File: GameArena.java (hunk 2 above)
Comment (finneyj) [reply GameArena.java:306]: so it does - cool.

File: GameArena.java [hunk 5]
```diff
@@ -451,13 +452,23 @@ public void removeText(Text t)
...
+	/**
+	 * Pause for the configured milliseconds using {@link GameArena#setSleepMillis(int)}. 
 	 * This method causes your program to delay for 1/50th of a second. You'll find this useful if you're trying to animate your application.
 	 *
 	 */
 	public void pause()
 	{
-		this.pause(20);
+		this.pause(this.sleepMillis);
 	}
```
Comment (finneyj) [inline GameArena.java:509]: Ta. Naming is not very intuitive though. Kinda want to go with 

```
setFrameRate(int milliseconds)
```

but that also implies we shoukd be using frequency and not period.

setFramePeriod() ? set frameDelay? setPauseDelay()?

Also, a matching accessor please, and make pause(ms) private?

Comment (danorris709) [reply GameArena.java:509]: Alright I'm not sure what the most intuitive name would be. Probably setFrameDelay or setPauseDelay as that makes the most sense given what it does.

But doesn't making pause(ms) private also defeat the point? Perhaps someone wants to pause the thread for an atypical amount of time?

Comment (finneyj) [reply GameArena.java:509]: also see line 143. We should probably set that to be half of the inter frame spacing in use.

Comment (finneyj) [reply GameArena.java:509]: Let's go with setPauseDelay.

I think pause(ms) being private is the way to go, just for simplicity. Then it encourages users to instead use setPauseDelay() and pause(), which will result in simpler code for users in the general case.

I'm struggling to think of a reason why using a pause() that's not related to your frame rate, and also it uses method overloading (and i'm trying to keep the API consistent with topcis we've covered in the course)

Comment (danorris709) [reply GameArena.java:509]: Alright, that sounds fairly logical to me. I can't see any reason for anyone to use pause(ms) either and if they're desperate I suppose they can always either create a new fork or use some sort of hacky reflection.

File: GameArena.java [hunk 6]
```diff
@@ -457,28 +457,28 @@ public void removeText(Text t)
 	 * 
 	 * @return The millis for which the thread will sleep between each frame
 	 */
-	public int getFrameDelay() {
-		return this.frameDelay;
+	public int setPauseDelay() {
+		return this.pauseDelay;
```
Comment (finneyj) [inline GameArena.java:488]: slow down my friend...

getPauseDelay please. :)

File: GameArena.java (hunk 5 above)
Comment (finneyj) [reply GameArena.java:509]: yep - or use another timer. :)

p.s. could you replace millis with milliseconds please - SI units rule.

File: GameArena.java [hunk 7]
```diff
@@ -339,6 +339,8 @@ private void addThing(Object o, int layer)
 						this.things.add(position, o);
 						added = true;
 					}
+
+					position++;
 				}
```
Comment (finneyj) [inline GameArena.java:359]: ok..... how much have you actually tested this patch?

Comment (danorris709) [reply GameArena.java:359]: I was testing it when I came across this. Definitely should've done before making the PR. My bad.

**Pull request closed without merging**